File: src/main/java/ai/magicdb/plugin/dm/config/DataServiceConfig.java
package ai.magicdb.plugin.dm.config;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

/**
 * 数据服务配置类
 * 确保控制器被正确扫描
 */
@Configuration
@ComponentScan(basePackages = {
    "ai.magicdb.plugin.dm.controller"
})
public class DataServiceConfig {
    // 配置类，用于确保控制器被正确扫描
}

File: magicdb-server/magicdb-data-service/magicdb-data-service-core/src/main/kotlin/ai/magicdb/data/service/core/config/DataServiceConfig.kt
package ai.magicdb.data.service.core.config

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.cache.annotation.EnableCaching
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.core.JdbcTemplate
import javax.sql.DataSource

/**
 * Configuration for data service module
 */
@Configuration
@EnableCaching
@ComponentScan(basePackages = ["ai.magicdb.data.service"])
class DataServiceConfig {
    
    /**
     * Create JdbcTemplate bean
     */
    @Bean
    fun jdbcTemplate(dataSource: DataSource): JdbcTemplate {
        return JdbcTemplate(dataSource)
    }
    
    /**
     * Create ObjectMapper bean if not already defined
     */
    @Bean
    fun objectMapper(): ObjectMapper {
        return ObjectMapper()
    }
}
